Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,453 @@
+                                              DeviceEventCategory == 8388608, "Expanded - Unified Policy Engine (8388608)",
+                                              "Legacy Category Not Mapped"
+                                              )
+  | extend sosIPSPriority = case(ipspri == 1, "High (1)",
+                                              ipspri == 2, "Medium (2)",
+                                              ipspri == 3, "Low (3)",
+                                              ""
+                                              )
+  | extend sosAntiSpywarePriority = case(spypri == 1, "High (1)",
+                                              spypri == 2, "Medium (2)",
+                                              spypri == 3, "Low (3)",
+                                              ""
+                                              )
+  | extend
+      EventVendor = "SonicWall"
+    , EventProduct = "Firewall"
+    , sosDeviceModel = DeviceProduct
+    , DvcOs = "SonicOS"
+    , DvcOsVersion = EventProductVersion
+    , DvcIdType = "Other"
+    , Dvc = sosSerialNumber
+    , DvcDescription = DeviceProduct
+    , ASimMatchingHostname = "-"
+    , ASimMatchingIpAddr = "-"
+    , NetworkIcmpType = tostring(NetworkIcmpType)
+    , NetworkIcmpCode = toint(NetworkIcmpCode)
+    , NetworkBytes = tolong(coalesce(toint(ReceivedBytes), 0) + coalesce(toint(SentBytes), 0))
+    , FieldDeviceCustomNumber1Label = column_ifexists("FieldDeviceCustomNumber1Label", "") // FieldDeviceCustomNumberXLabel (cnXLabel=)
+    , FieldDeviceCustomNumber2Label = column_ifexists("FieldDeviceCustomNumber2Label", "") // FieldDeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber1Label = column_ifexists("DeviceCustomNumber1Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber2Label = column_ifexists("DeviceCustomNumber2Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber3Label = column_ifexists("DeviceCustomNumber3Label", "") // Connection Duration in milliseconds (for Connnection Closed msgs)
+    , fw_action = column_ifexists("fw_action", "")
+    , sosFirewallAction = column_ifexists("fw_action", "")
+    , susr = column_ifexists("susr", "") // Logged-in username associated with the log event.
+    , af_polid = column_ifexists("af_polid", "")
+    , af_action = column_ifexists("af_action", "")
+    , af_policy = column_ifexists("af_policy", "")
+    , af_service = column_ifexists("af_service", "")
+    , af_type = column_ifexists("af_type", "")
+    , af_object = column_ifexists("af_object", "")
+    , contentObject = column_ifexists("contentObject", "") // App Firewall/App Rule Content.
+    , srcV6 = column_ifexists("srcV6", "") // Source IPv6 IP
+    , dstV6 = column_ifexists("dstV6", "") // Destination IPv6 IP
+    , appName = column_ifexists("appName", "") // Non-signature Application name that matches the Application ID "app" or "f".

```

Comment:
we have DstAppName, SrcAppName. appname not fitting for these?

Reply:
These fields are a bit too narrowly scoped for our logging. We have App Signatures categories by App within Categories. Each Signature, App, and Category has a name and ID number. It is difficult to fit all of these values into just a couple of fields. I also find that some fields require specific values that may not fit the broader scope of our services. With that said, I'm working on using some of these fields. I will likely still need to store some information in AdditionalFields that seems redundant but I will try to keep it to as few as possible.